### DISCUSSION TITLE
WiX Life Cycle Policy

### DISCUSSION THREAD

**Discussion by joeloff** [wixtoolset/issues#7991] [Questions] [answered]
Hi

First, thank you for continued support of WiX v3.

With the recent announcement of the EOL date for WiX 3, I was wondering whether there was an official life cycle policy anywhere. Is WiX 4 considered a long term servicing release, and if so for how long? When 5 hits RTM, does that start a clock against 4, for example 4 remains in support for N years after 5 RTMs?

I'm trying to do some planning and deciding whether 4 or 5 would be the next logical jump.

Thanks,
Jacques

**Comment by robmen** [comment, marked as answer]
Good question. I'm still working on final details and this security issue has helped refine my thinking. [You can read my early thoughts on the topic](https://github.com/orgs/wixtoolset/discussions/7446) but here are a couple of items that are coming into focus for me.

1. Guarantees (beyond those specified in the MS-RL) are only made to [Customers via their support contract](https://www.firegiant.com/services/support-programs/)
   * For example, all dot releases of WiX v4 addressed issues that impacted a Customer
   * If you are a Customer, you can stop reading here, you are taken care of
  
2. We take our role in maintaining the security of the ecosystem we participate in seriously, so:
   * We will fix security issues found in versions released within `N` months of the last major version
   * This provides a window of time for Consumers to be secure while they are migrating to the latest version
   * We haven't decided on `N` but we will pick something like 3, 6, 9 or 12.
   * For example, if `N` is 6 and WiX v5 releases April 5, 2024 (as planned) then WiX v4 will get security fixes ported to it and generally released through October 5, 2024.

3. Consumers should plan to adopt the `-rc.#` releases as they are made available
   * Regressions found during the -rc process are _far_ more likely to be fixed than waiting to try out RTM

> Aside: WiX v3 is an outlier from the above bullets for several reasons (WiX v4's long development time being a big one). That is why WiX v3 is getting a 22 month migration window.

**Comment by joeloff** [comment]
This is very helpful Rob, thank you.
